analyzers/src/SonarAnalyzer.CSharp/Wrappers/ObjectCreationFactory.cs

```diff
@@ -31,6 +31,7 @@ public interface IObjectCreation
         ArgumentListSyntax ArgumentList { get; }
         ExpressionSyntax Expression { get; }
         string TypeAsString(SemanticModel semanticModel);
+        ISymbol TypeSymbol(SemanticModel semanticModel);
```

Comment:
Aren't there one or two other recent PRs that can benefit from this functionality as well?

Reply:
Many of them use CSharpObjectInitializationTracker.IsTrackedType, which is also used by non-object creation assignments to check if the type is being tracked. Fixed in a few places I could